src/utils/package.ts

```diff
@@ -0,0 +1,50 @@
+export type PackageDotJson = {
+  scripts?: Record<string, string>;
+  dependencies?: Record<string, string>;
+  devDependencies?: Record<string, string>;
+};
+
+type NpmPackage = {
+  name: string;
+  version: string;
+};
+
+/**
+ * Checks if @param packageJson has any of the @param packageNamesList package names
+ * listed as a dependency or devDependency.
+ * If so, it returns the first package name that is found, including the
+ * version (range) specified in the package.json.
+ */
+export function findPackageFromList(
+  packageNamesList: string[],
+  packageJson: PackageDotJson,
+): NpmPackage | undefined {
+  const installedSdkPackages = packageNamesList
+    .map((packageName) => ({
+      name: packageName,
+      version: getPackageVersion(packageName, packageJson),
+    }))
+    .filter((sdkPackage): sdkPackage is NpmPackage => !!sdkPackage.version);
```

Comment:
l: We could think about using `Array.find` here instead. Feel free to skip and ignore.

Reply:
Oh god, what did I do here 😂